config/routes.rb

```diff
@@ -44,14 +44,16 @@
     end
   end
 
-  resources :trainees, except: :destroy do
+  resources :trainees do
     scope module: :trainees do
       resource :training_details, concerns: :confirmable, only: %i[edit update], path: "/training-details"
       resource :programme_details, concerns: :confirmable, only: %i[edit update], path: "/programme-details"
       resource :contact_details, concerns: :confirmable, only: %i[edit update], path: "/contact-details"
       resource :trainee_id, concerns: :confirmable, only: %i[edit update], path: "/trainee-id"
       resource :start_date, concerns: :confirmable, only: %i[edit update], path: "/trainee-start-date"
 
+      get "/confirm-delete", to: "confirm_delete#show", as: :show_confirm_delete
```

Comment:
I think you can also get rid of the `as: :show_confirm_delete` here.

The automatically generated helper method will then be `trainee_confirm_delete_path`, which I think is nice and clear.

You don't normally have the word 'show' in the method even though that's the name of the method in the controller (e.g. it's `trainee_path(trainee)` rather than `trainee_show_path(trainee)`).